python/tskit/trees.py

```diff
@@ -357,6 +357,19 @@ def __eq__(self, other):
             and self.metadata == other.metadata
         )
 
+    def alleles(self):
+        """
+        Return the set of all the alleles defined at this site (the ancestral state and
+        the derived states of any mutations).
+
+        .. note::
+            Unlike the .alleles attribute of the :class:`Variant` class, this does not
+            include ``None`` as an allele when there is missing data.
+        """
+        alleles = {m.derived_state for m in self.mutations}
+        alleles.add(self.ancestral_state)
+        return alleles
```

Comment:
A matter of taste so I don't feel strongly, but this reads better to my eye.

`return {m.derived_state for m in self.mutations} | {self.ancestral_state}`